java-api/src/main/java/com/sap/cloud/security/x509/X509Parser.java

```diff
@@ -0,0 +1,132 @@
+	 * @param certificate
+	 *            Base64 encoded X509 certificate
+	 * @return Thumbprint of X509 certificate
+	 * @throws NoSuchAlgorithmException
+	 *             when a particular cryptographic algorithm is requested but is not
+	 *             available in the environment
+	 * @throws CertificateException
+	 *             is thrown if errors occur while decoding X509 certificate
+	 */
+	public static String getX509Thumbprint(String certificate) throws NoSuchAlgorithmException, CertificateException {
+		X509Certificate x509Certificate = decodeX509(
+				Objects.requireNonNull(certificate, "X509 certificate can't be null"));
+		MessageDigest sha256Digest = MessageDigest.getInstance("SHA-256");
+		byte[] hashedX509 = sha256Digest.digest(x509Certificate.getEncoded());
+		String base64UrlEncoding = Base64.getUrlEncoder().encodeToString(hashedX509);
+		LOGGER.debug("Base64-url encoded thumbprint: {}", base64UrlEncoding);
+		return normalizeThumbprint(base64UrlEncoding);
+	}
+
+	/**
+	 * Thumbprint generation alternative to {@link #getX509Thumbprint(String)}
+	 * without decoding into {@link X509Certificate}class
+	 *
+	 * @see <a href="https://tools.ietf.org/html/rfc8705#section-3.1"></a>
+	 * @param base64certificate
+	 *            certificate encoded in Base64, supports also PEM encoded
+	 *            certificates
+	 * @return Thumbprint of X509 certificate
+	 * @throws NoSuchAlgorithmException
+	 *             when a particular cryptographic algorithm is requested but is not
+	 *             available in the environment
+	 */
+	public static String getCertificateThumbprint(String base64certificate) throws NoSuchAlgorithmException {
+		base64certificate = base64certificate
+				.replace(BEGIN_CERTIFICATE, "")
+				.replace(END_CERTIFICATE, "")
+				.replaceAll("\\s", "");
+		MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+		byte[] der = Base64.getDecoder().decode(base64certificate.getBytes(StandardCharsets.UTF_8));
+		byte[] hashedX509 = sha256.digest(der);
+		return normalizeThumbprint(Base64.getUrlEncoder().encodeToString(hashedX509));
+	}
+
+	/**
+	 * The base64url-encoded value MUST omit all trailing pad '=' characters and
+	 * MUST NOT include any line breaks, whitespace, or other additional characters
+	 *
+	 * @param thumbprint
+	 *            raw thumbprint of X509
+	 * @return thumbprint without trailing '=' and '\n'
+	 */
+	static String normalizeThumbprint(String thumbprint) {
+		return thumbprint.trim().replaceFirst("=*$", "").replace("\n", "").replace("\\n", "");
+	}
+
+	/**
+	 * Decodes X509 base64 encoded certificate into X509Certificate class
+	 * 
+	 * @param encodedX509
+	 *            base64 encoded X509 certificate
+	 * @return Decoded X509Certificate
+	 * @throws CertificateException
+	 *             if String value of certificate cannot be parsed
+	 */
+	private static X509Certificate decodeX509(@Nonnull String encodedX509) throws CertificateException {
```

Comment:
this method is called parseCertString in the go client lib

Reply:
I can rename it to parseCert() but I don't think String is required in the method name